kupfer/plugin/id2/__init__.py

```diff
@@ -0,0 +1,195 @@
+""" ID2 – Guess Identifier (Kupfer Plugin). See https://id2.dev
+"""
+__kupfer_name__ = _("Identifier Resolver")
+__kupfer_sources__ = ()
+__kupfer_text_sources__ = ("IdentifierSource",)
+__kupfer_actions__ = ("LookupID",)
+__description__ = _(
+    """Detect and open DOIs, ISBNs, OrcIDs, crypto-addresses, geo-coordinates, and more in your web-browser. Examples:
+
+- Unicode:  \t"U+1F194"
+- UnixTime: \t"1606324253"
+- Location: \t"8FWH4HX8+QR"
+- DOI:    \t\t"10.1109/FDL.2018.8524068"
+- OrcID:    \t"0000-0002-0006-7761"
+- BitCoin:  \t"3FZbgi29cpjq2GjdwV8eyHuJJnkLtktZc5"
+
+For a full list, see https://id2.dev
+
+To disable a certain ID, write a single character (e.g. "x") into the URL field below.
+"""
+)
+__version__ = "2022.1"
+__author__ = "Emanuel Regnath"
+
+
+from kupfer import utils, plugin_support
+from kupfer.objects import Action, TextLeaf, TextSource
+import re
+import json
+import sys
+import os
+import urllib.parse
+
+
+# global table holding identifier classes and icon names
+CLASS_NAMES = {
+    "i": ["Identifier", "edit-find"],
+    "g": ["Geo-Location", "applications-internet"],
+    "d": ["Document", "x-office-document"],
+    "o": ["Object", "package-x-generic"],
+    "h": ["Hash", "dialog-password"],
+    "t": ["Time", "x-office-calendar"],
+    "w": ["Person", "stock_person"],
+}
+
+# global data
+id2data = json.load(
+    open(os.path.join(os.path.dirname(__file__), "id2data.json"), "r"))
+
+
+# settings are auto-generated by "id2-settings-generator.py"
+__kupfer_settings__ = plugin_support.PluginSettings(
+    {
+        "key": "i_ip4",
+        "label": "IP4",
+        "type": str,
+        "value": "https://whois.domaintools.com/",
+    },
+    {
+        "key": "i_ip6",
+        "label": "IP6",
+        "type": str,
+        "value": "https://whois.domaintools.com/",
+    },
+    {
+        "key": "i_utf8",
+        "label": "UTF8",
+        "type": str,
+        "value": "https://www.compart.com/en/unicode/U+",
+    },
+    {
+        "key": "g_gps",
+        "label": "GPS",
+        "type": str,
+        "value": "https://www.google.com/maps/search/",
+    },
+    {
+        "key": "o_ean",
+        "label": "EAN",
+        "type": str,
+        "value": "https://www.ean-search.org/?q=",
+    },
+    {
+        "key": "o_upc",
+        "label": "UPC",
+        "type": str,
+        "value": "https://www.ean-search.org/?q=",
+    },
+    {"key": "o_asin", "label": "ASIN", "type": str, "value": "https://amzn.com/"},
```

Comment:
look's like not formatted by black ;)

Reply:
Yes this is black! I have manually written every key in a separate line but black insists on writing everything in one line... I would also like to have it more consistent...